### PR TITLE
feat(accounts): implement account register and change password

### PR DESCRIPTION
**Pull request by noice-noise**
### Frontend
- Added common shared Components
- Added basic Navigation
- Added Accounts Register & Success pages
- Added simple route protection
- Added Change Password page (nested in Settings route)
- Added first login Change Default Password prompt page
- Added simple Unauthorized page display

## Checklist

_First, create a draft pull request. Then mark the PR as ready for review when all necessary checkbox items has been completed_

- [x] Reviewed own code
- [x] Commented on code that is hard to understand
- [ ] Implemented tests for the feature/bugfix
- [x] All GitHub status checks pass
- [x] The frontend and backend PR previews have been deployed

**Backend only**

- [ ] Added test data for new entity
- [ ] Generated the client api if there are new or deleted endpoints and/or DTOs

### CODE REVIEW

File: frontend/src/shared/schemas/validation/email.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import * as yup from 'yup';
+
+export const email = yup
```
Comment (princhcanal) [inline frontend/src/shared/schemas/validation/email.ts:3]: i think we should give the validators more descriptive names. maybe something like `emailValidator`?

File: frontend/src/shared/components/nav/Nav.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+export const Nav = () => {
+  return (
+    <Flex as='nav'>
+      <NavLink label='Home' route='/' />
+      <NavLink label='Accounts' route='/accounts' />
+      <NavLink label='Events' route='/events' isDisabled />
+      <NavLink label='Venues' route='/venues' isDisabled />
+      <NavLink label='Departments' route='/departments' isDisabled />
+      <NavLink label='Equipment' route='/equipment' isDisabled />
+      <NavLink label='Information' route='/information' isDisabled />
```
Comment (princhcanal) [inline frontend/src/shared/components/nav/Nav.tsx:13]: when i click the disabled routes, they still navigate to the page and show a 404 error page. no need to change as it doesn't really matter but just something i noticed

**Review by princhcanal — APPROVED** [review]
very noice pr 🚀 it's nice to finally have some real components in our app and you did great on  extracting the reusable components and layouts 💜 

there's just some minor things that i commented that would be nice if you could address if you agree. if not, feel free to merge :)

also, no need to change this either but the branch name doesn't have the `feature/` prefix. but you don't need to change it since i think the pr would break if you do

**Comment by princhcanal** [comment]
some more things i noticed: 

- when we change password on first login, the user should be directed to the home page but it stays in the first login page. even when i try to change the route manually. is this a backend issue?
- i think we shouldn't show the `Accounts` link in the navbar if the user doesn't have permissions

**Comment by noice-noise** [comment]
> some more things i noticed:
> 
> * when we change password on first login, the user should be directed to the home page but it stays in the first login page. even when i try to change the route manually. is this a backend issue?
> 
> > Update: this shouldn't be a backend issue since the returned user after changing passwords already has `isFirstLogin` set to false
> > Update 2: it is a backend issue, however, that we return the hashed password but i will fix this :)
> 
> Screen Shot 2022-09-03 at 3 27 26 PM
> 
> * imo, i think we shouldn't show the `Accounts` link in the navbar if the user doesn't have permissions. do you agree?

I think there's also an issue in the frontend side. It seems that the user data in the localStorage has not been updated after any changes. Let me look into this more.

**Comment by noice-noise** [comment]
> some more things i noticed:
> 
> * when we change password on first login, the user should be directed to the home page but it stays in the first login page. even when i try to change the route manually. is this a backend issue?
> 
> > Update: this shouldn't be a backend issue since the returned user after changing passwords already has `isFirstLogin` set to false
> > Update 2: it is a backend issue, however, that we return the hashed password but i will fix this :)
> 
> Screen Shot 2022-09-03 at 3 27 26 PM
> 
> * imo, i think we shouldn't show the `Accounts` link in the navbar if the user doesn't have permissions. do you agree?

I agree that we need to implement such protection soon. But for now, can we add a separate Jira issue on this one? 
As I wanted to discuss with the team further our direction on feature protection in our next sprint planning.

**Comment by noice-noise** [comment]
> very noice pr 🚀 it's nice to finally have some real components in our app and you did great on extracting the reusable components and layouts 💜
> 
> there's just some minor things that i commented that would be nice if you could address if you agree. if not, feel free to merge :)
> 
> also, no need to change this either but the branch name doesn't have the `feature/` prefix. but you don't need to change it since i think the pr would break if you do

Thanks you very much for such a detailed PR review 💜🚀

**Comment by princhcanal** [comment]
> I agree that we need to implement such protection soon. But for now, can we add a separate Jira issue on this one? As I wanted to discuss with the team further our direction on feature protection in our next sprint planning.

yes, could you add the jira issue for that? thanks :)

**Review by princhcanal — APPROVED** [review]
thanks for implementing the changes. should be ready to merge now :)

**Comment by noice-noise** [comment]
> > I agree that we need to implement such protection soon. But for now, can we add a separate Jira issue on this one? As I wanted to discuss with the team further our direction on feature protection in our next sprint planning.
> 
> yes, could you add the jira issue for that? thanks :)

Sure, sure!

**Pull request merged**